extensions/ql-vscode/test/unit-tests/databases/db-manager.test.ts

```diff
@@ -362,6 +369,137 @@ describe("db manager", () => {
+        variantAnalysisList,
+      ]);
+      let expandedItems = await app.workspaceState.get<ExpandedDbItem[]>(
+        DbManager.DB_EXPANDED_STATE_KEY,
+      );
+      expect(expandedItems?.length).toEqual(1);
+
+      // Remove item from expanded state
+      const dbItem = createVariantAnalysisUserDefinedListDbItem({
+        listName,
+      });
+
+      await dbManager.removeDbItemFromExpandedState(dbItem);
+      expandedItems = await app.workspaceState.get<ExpandedDbItem[]>(
+        DbManager.DB_EXPANDED_STATE_KEY,
+      );
+
+      expect(expandedItems?.length).toEqual(0);
+    });
+
+    it("should rename item in expanded state", async () => {
+      // Add item to config
+      const listName = "my-list-3";
+      const dbConfig = createDbConfig({
+        remoteLists: [{ name: listName, repositories: [] }],
+      });
+      await saveDbConfig(dbConfig);
+
+      // Add item to expanded state
+      const variantAnalysisList = {
+        kind: ExpandedDbItemKind.RemoteUserDefinedList,
+        listName,
+      };
+
+      await app.workspaceState.update(DbManager.DB_EXPANDED_STATE_KEY, [
+        variantAnalysisList,
+      ]);
+      let expandedItems = await app.workspaceState.get<ExpandedDbItem[]>(
+        DbManager.DB_EXPANDED_STATE_KEY,
+      );
+      expect(expandedItems?.length).toEqual(1);
+      let expandedItem =
+        expandedItems![0] as VariantAnalysisUserDefinedListExpandedDbItem;
+      expect(expandedItem.listName).toEqual(listName);
+
+      // Rename item
+      const dbItem = createVariantAnalysisUserDefinedListDbItem({
+        listName,
+      });
+
+      await dbManager.renameList(dbItem, "new-list-name");
+      expandedItems = await app.workspaceState.get<ExpandedDbItem[]>(
+        DbManager.DB_EXPANDED_STATE_KEY,
+      );
+
+      expect(expandedItems?.length).toEqual(1);
+      expandedItem =
+        expandedItems![0] as VariantAnalysisUserDefinedListExpandedDbItem;
+      expect(expandedItem.listName).toEqual("new-list-name");
+    });
+
+    it("should remove non existent items in expanded state when item is expanded", async () => {
+      // We remove items from the expanded state if they are not in the config
+
+      // Populate expanded state with non existent item
+      const listName = "my-list-4";
+      const variantAnalysisList = {
+        kind: ExpandedDbItemKind.RemoteUserDefinedList,
+        listName,
+      };
+
+      await app.workspaceState.update(DbManager.DB_EXPANDED_STATE_KEY, [
+        variantAnalysisList,
+      ]);
+      let expandedItems = await app.workspaceState.get<ExpandedDbItem[]>(
+        DbManager.DB_EXPANDED_STATE_KEY,
+      );
+      expect(expandedItems?.length).toEqual(1);
+      const expandedItem =
+        expandedItems![0] as VariantAnalysisUserDefinedListExpandedDbItem;
+      expect(expandedItem.listName).toEqual(listName);
+
+      // Trigger adding an item that is not in the config
```

Comment:
I think it would be more realistic if the test added a different item (that exists), and checked that the other one (that doesn't exist) was removed.

Reply:
👍 Good idea

Reply:
(Optional) I think this is slightly confusing because there are 3 different lists at play. I would do the following:
- Have expanded state have only 1 item (the `removedVariantAnalysisList`)
- Trigger adding a new item in the config (the `variantAnalysisList`)
- Check that the expanded state no longer contains the `removedVariantAnalysisList`